Compass/LoginViewController.swift

```diff
@@ -0,0 +1,216 @@
+        
+        
+        passTextField.placeholder = "password..."
+        passTextField.text = "pass"
+        passTextField.textColor = UIColor.white
+        passTextField.isSecureTextEntry = true
+        passTextField.layer.backgroundColor = UIColor.blue.cgColor
+        passTextField.attributedPlaceholder = NSAttributedString(string: "Password...", attributes: [NSAttributedStringKey.foregroundColor : UIColor.white])
+        
+        
+        let stackView = UIStackView(arrangedSubviews: [titleLabel, NametextField, passTextField, loginButton, switchPageButton])
+        stackView.axis = .vertical
+        stackView.spacing = 10;
+        
+        if isloginPage {
+            titleLabel.text = "Login Page"
+            loginButton.setTitle("Login", for: .normal)
+            switchPageButton.setTitle("Go to Signup page", for: .normal)
+            
+        } else {
+            titleLabel.text = "Signup Page"
+            loginButton.setTitle("Signup", for: .normal)
+            switchPageButton.setTitle("Go to Login page", for: .normal)
+        }
+        
+        self.view.addSubview(stackView)
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([stackView.widthAnchor.constraint(equalToConstant: 200),
+                                     stackView.centerXAnchor.constraint(lessThanOrEqualTo: self.view.centerXAnchor),
+                                     stackView.centerYAnchor.constraint(lessThanOrEqualTo: self.view.centerYAnchor)])
+        
+        
+        
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    @objc private func loginButtonPressed() {
+        
+        if NametextField.text == "" || passTextField.text == "" {
+            createAlert(title: "Invalid username or password format", message: "Please enter valid username or password")
+        } else if !isValidEmail(testStr: NametextField.text!) {
+            createAlert(title: "Invalid e-mail address", message: "Please enter valid e-mail")
+        } else {
+            
+            activityIndicator = UIActivityIndicatorView(frame: CGRect.zero)
+            activityIndicator.center = self.view.center
+            activityIndicator.hidesWhenStopped = true
+            activityIndicator.activityIndicatorViewStyle = UIActivityIndicatorViewStyle.gray
+            view.addSubview(activityIndicator)
+            activityIndicator.startAnimating()
+            UIApplication.shared.beginIgnoringInteractionEvents()
+        
+        
+        if isloginPage {
+            
+            PFUser.logInWithUsername(inBackground: NametextField.text!, password: passTextField.text!) { (user, error) in
+                self.activityIndicator.stopAnimating()
+                UIApplication.shared.endIgnoringInteractionEvents()
+            
+                if error != nil {
```

Comment:
I can't remember what we changed, but this is another good candidate for a guard statement